assignment-2/lean.md

```diff
@@ -1,3 +1,96 @@
+
+The concept of waste is a traversal concept in other software
+development methodologies. *RUP* (Rational Unified Process) focuses on
+reducing risks and scrum focuses on delivering product. Both of them do
+not pay special attention on waste.
+
+On one hand, RUP can be more wasteful that scrum. The early stages of
+RUP (business modelling, requirements and analysis & design) take a long
+period and effort and, although they produce deliverables, they are not
+the one requested by the client. This qualifies as *muda* type I, as it
+is a waste, but it is necessary. This waste does not exist on scrum. In
+scrum, the documentation is almost not existing. This means that the
+team does not produce any other deliverables but the product that
+the client requests.
+
+RUP also have more phases besides development. The early phases focus on
+having clear requirements and late ones focus on deployment and testing.
+The division of the phases aims to reduce the risks in the project by
+specifying the details of the project ahead and making sure that the
+product has been properly deployed and tested. Although necessary, these
+phases add extra waste in terms of processes and bureaucracy, instead of
+being integrated within the development. In scrum, this waste does not
+exist because it is considered part of the development and it does not
+carry extra overhead due to being integrated into the development phase.
+
+On the other hand, scrum can be more wasteful than RUP. Scrum works with
+smalls teams and big projects require more than one scrum team working
+on it. This means that those team would have each one their own *scrum
+master* and *product owner* who will talk with the *stakeholders*. This
+situation will lead to role duplicity which is not optimal at all. RUP
+only has one management team which is an improvement in terms of wasting
+over scrum.
+
+Scrum also focuses on small iterations of the product: features are
+added iteration after iteration. The client then will not receive the
+final product requested during the first iterations, which is a waste
+because it is not what was requested. In addition, those iterations may
+lead to having to reimplement features due to changes in the
+architecture. This problem, which is inherited from the agile
+development philosophy, does not exist in RUP as it has only one
+development phase. Scrum also requires to perform testing every
+iteration, which RUP does only once.
+
+Scrum might deliver a product which is not the exact one requested.
+*Product owner* will discuss details with the *stakeholders* but during
+development and the discussions with the developers, some features might
+stay in the backlog and never make it into development, suffer changes
+or directly taken off from it. Usually those features are small details
+but they still differ from the originally requested product. RUP states
+that the final product delivered will be the one discussed during the
+design phases.
```

Comment:
So if you find out that the feature described is not what you want, you build it anyway? Talk about waste... ;P